src/javascripts/file_upload.js

```diff
@@ -0,0 +1,56 @@
+// Show file name, file upload status and check file size
+
+UK_Parliament.fileUploader = function () {
+
+  var fileInput = document.querySelector('[data-file-upload="button"]');
+
+  if (fileInput) {
+
+    // Local variables
+    var
+      fileIcon = document.querySelector('[data-file-upload="icon"]'),
+      fileStatus = document.querySelector('[data-file-upload="status"]');
+
+    fileInput.onchange = function () {
+
+      // Grab file object reference, determine size and construct filename
+      var
+        fileObject = this.files,
+        fileObjectSize = fileObject[0].size,
+        fileName = this.value.split(/(\\|\/)/g).pop(),
+        fileNameFormatted = '<strong>\'' + fileName + '\'</strong>';
+
+      // Check for aria attribute and manage addition / class removal accordingly
+      fileStatus.hasAttribute('aria-live') ? fileStatus.classList.remove('message--error') : fileStatus.setAttribute('aria-live', 'assertive');
+
+      // Check if file exceeds size limit (currently set to 2MB)
+      if (fileObjectSize > 2097152) {
+        fileStatus.classList.add('message--error');
+        fileStatus.innerHTML = 'The file you have uploaded is too big. Please make sure it is smaller than 2MB.';
+        fileIcon.setAttribute('data-file-upload', 'failure');
```

Comment:
Couldnt we add class `svg--file__failure` and remove the additional defined classes in `src/stylesheets/elements/_icons.scss`